torch/distributions/binomial.py

```diff
@@ -113,11 +113,9 @@ def log_prob(self, value):
         log_factorial_n = torch.lgamma(self.total_count + 1)
         log_factorial_k = torch.lgamma(value + 1)
         log_factorial_nmk = torch.lgamma(self.total_count - value + 1)
-        max_val = (-self.logits).clamp(min=0.0)
-        # Note that: torch.log1p(-self.probs)) = max_val - torch.log1p((self.logits + 2 * max_val).exp()))
+        # Note that: torch.log1p(-self.probs)) = - torch.log1p(self.logits.exp()))
         return (log_factorial_n - log_factorial_k - log_factorial_nmk +
-                value * self.logits + self.total_count * max_val -
-                self.total_count * torch.log1p((self.logits + 2 * max_val).exp()))
+                value * self.logits - self.total_count * torch.log1p(self.logits.exp()))
```

Comment:
In PyTorch, `logits != probs.exp()`, so this formula seems wrong to me. Previously, the tests passed, which makes me confused.

Reply:
@fehiepsi - That is correct - `logits != probs.log()` for the binary case. 

Does this seem reasonable?
```
value * probs.log() + (total_count - value) * torch.log1p(-probs))
 = value * (probs.log() - torch.log1p(-probs)) + total_count * torch.log1p(-probs)
 = value * logits + total_count * (1 - probs)
 = value * logits - total_count * torch.log1p(self.logits.exp()))
```

Reply:
first equation: correct
second equation: correct
third equation: incorrectly assumes `probs=logits.exp()`

Reply:
Thanks! It seems reasonable to me now. @fritzo will be confused. :D

Reply:
Just in case, this causes confusion for others, expanding on the last step:

```
(1 - probs).log()
 = (1 - (logits.exp()/(1 + logits.exp())).log()
 = (1 / (1 + logits.exp())).log()
 = - (1 + logits.exp())).log()
 = - log1p(logits.exp())
```